homework/tests/example.spec.js

```diff
@@ -17,15 +20,546 @@ const data = [
 
 data.forEach(version => {
   test.describe(version + ': Add', () => {
-    test('Concatenating 2 and 3 results in 23', async ({ page }) => {
-      await page.goto('https://testsheepnz.github.io/BasicCalculator');
-      await page.selectOption('#selectBuild', { label: version});
-      await page.locator('#number1Field').type('2');
-      await page.locator('#number2Field').type('3');
-      await page.selectOption('#selectOperationDropdown', {label: 'Add'});
-      await page.locator('#calculateButton').click();
-  
-      await expect(page.locator('#numberAnswerField')).toHaveValue('5');
+    test('Adding 2 and 3 results in 5', async ({ page }) => {
+      let searchPage = new SearchPage(page);
+      await searchPage.navigate();
+      await searchPage.selectBuild(version);
+
+      let calcForm = new CalcForm(page);
+      await calcForm.enterFirstNumber('2');
+      await calcForm.enterSecondNumber('3');
+      await calcForm.selectOperation('Add');
+
+      let ansForm = new AnsForm(page);
+      await ansForm.clickCalcButton();          
+      await expect(page.locator('#numberAnswerField' ), 'Should answer 5').toHaveValue('5');
```

Comment:
this assertion could also be moved to page object, to not repeat this line every time. If you want to keep the full assertion in test, at least locator can be parametrized and written in page object, so that the same value can be reused and not copied every time.

Reply:
I tried to move to the page object, but I got an error and just gave up.